### PR TITLE
[Fix][UI] Fix hive datasource doesn't show principal bug

### LINKED ISSUE #10436: [Bug] [DataSource] edit an exist hive datasource connection, don't show principal and keytab
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

when create hive datasource connection, have principal and keytab input item,but if we edit the hive datasource connection, it can't show pirncipal and keytab. One more thing, if we click create hive datasource first, then click edit exist hive datasource, it will show principal.
image

image

### What you expected to happen

it can show principal and keytab when edit hive datasource connection.

### How to reproduce

1、create an hive datasource connection(if you have one, skip this)
2、click other like Home tabs, then click datasource tabs again
3、edit the hive datasource connection created in step 1.
4、you will find it don't show principal and keytab

### Anything else

_No response_

### Version

3.0.0-beta-1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by devosend**
## Purpose of the pull request

close #10436 
## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java [hunk 1]
```diff
@@ -339,7 +339,7 @@ public Result authedDatasource(@ApiIgnore @RequestAttribute(value = Constants.SE
     @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
     public Result getKerberosStartupState(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser) {
         // if upload resource is HDFS and kerberos startup is true , else false
-        return success(Status.SUCCESS.getMsg(), CommonUtils.getKerberosStartupState());
+        return success(Status.SUCCESS.getMsg(), true);
```
Comment (SbloodyS) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java]: The parameter `principal` and `KerberosStartupState` are strongly correlated. I don't think this needs to be modified. We should add a description to the document to tell users this.

Comment (devosend) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java]: This is a code that should not be submitted.

**Review by Amy0104 — APPROVED** [review]
LGTM

**Review by songjianet — APPROVED** [review]
LGTM

**Pull request merged**